test.py

```diff
@@ -140,69 +142,83 @@ class A(SEM):
 
 
 def test_sparse_map():
-    a = SEM("a", names="b c e")
+    a = SparseEnumap("a", names="b c e")
     assert (a.map(*"1 3".split(), c="2.2") ==
             OrderedDict([("b", "1"), ("c", "2.2"), ("e", None)]))
 
 
 def test_sparse_casted_tuple():
-    a = SEM("a", names="a b c e")
+    a = SparseEnumap("a", names="a b c e")
     a.set_types(to_int, to_int, float, float)
     casted_tuple = a.tuple_casted(*"1.1 2.2 3.3".split())
     assert casted_tuple == (1, 2, 3.3, None)  # missing values aren't casted
 
 
 def test_sparse_casted_tuple_with_default():
-    a = SEM("a", names="a b c e")
+    a = SparseEnumap("a", names="a b c e")
     a.set_types(to_int, to_int, float, int)
     a.set_defaults(e="HOOP")
     casted_tuple = a.tuple_casted(*"1.1 2.2".split())
     assert casted_tuple == (1, 2, None, "HOOP")  # missing values aren't casted
 
 
 def test_sparse_casted_map():
-    a = SEM("a", names="a b c e")
+    a = SparseEnumap("a", names="a b c e")
     a.set_types(to_int, to_int, float, float)
     casted_map = a.map_casted(*"1.1 2.2 3.3".split())
     assert tuple(casted_map.values()) == (1, 2, 3.3, None)
 
 
+def test_type_cast_exception():
+    """Make sure our type casting exceptions are informative"""
+    class A(Enumap):
+        a: int = auto()
+        this_here_is_a_bad_key: int = auto()
+
+    assert A.tuple_casted("1", "2") == (1, 2)
+    with pytest.raises(TypeCastError) as e:
+        A.tuple_casted("1", None)
+    assert "'this_here_is_a_bad_key' got invalid value 'None'" in str(e)
+    assert "of type <class 'NoneType'>" in str(e)
+    assert e.value.key == "this_here_is_a_bad_key"
```

Comment:
Inspired by #4, consider adding tests for the sparsely annotated and typeless case.